zenoh/src/api/publisher.rs

```diff
@@ -533,10 +547,10 @@ impl MatchingStatus {
     ///     .matching_status()
     ///     .await
     ///     .unwrap()
-    ///     .matching_subscribers();
+    ///     .matching();
     /// # }
     /// ```
-    pub fn matching_subscribers(&self) -> bool {
+    pub fn matching(&self) -> bool {
```

Comment:
Does it give any real advantage ? To me it seems it would complicate the internal code, fetching status for not much benefit.  Besides the way it is done simplifies porting to zenoh-c (and eventually to zenoh-pico) since it does not require to make identical copies of almost similar structs.

Reply:
Fine. Please loop with the API team and anyone concerned to validate everybody agrees with the API change.